feature/2-betterauth

rdgonzaga opened the pull request:
### Summary

Integrate BetterAuth (restricted to @dlsu.edu.ph) and initialize the auth DB schema. Implement auth service and backend session utilities.

closes \#2

### Type of Change

- [x] feat: New feature
- [ ] fix: Bug fix
- [x] docs: Documentation update
- [ ] chore/refactor: Maintenance or code restructure

### Notes for Reviewers

- removed unused folders under /features

rdgonzaga committed:
feat(auth): add auth instance and init auth schema

rdgonzaga committed:
feat(auth) add authservice, types, session utilities, and helpers

rdgonzaga committed:
chore(cleanup): remove unused feature folders

xandreiAThome commented on src/features/auth/services/service.ts:
Function based dapat like my earlier comment sa last pr. And di dapat direclty interface to the orm to access auth related entities let better auth handle it.

rdgonzaga replied:
okay okay noted i forgot again sorry since ginaya ko lang yung nakita ko sa documentation

rdgonzaga replied:
medyo confused ako sa second sentence, ikeep ko ba yung drizzle queries? or dapat iremove ko na lang tong user service tas gamitin ko lang yung session utilities functions ?

xandreiAThome replied:
instead of using drizzle queries gamitin mo yung api ng better auth to interface with the db, May equivalent functions yung better auth for those queries.

rdgonzaga replied:
im still confused, i cant find equivalent functions for getByUserId, getUserByEmail and updateUser. auth.api.getSession() lang yung nahanap ko. gamitin ko lang ba dapat yung session utilities functions (getsession, requireAuth, and getcurrentUser). 

if di naman siguro need yung getUserById/getUserByEmail/updateUser, thats good na (?). nilagay ko lang siya kasi nakakita ako na may nagimplement ng ganyan while studying it kanina

xandreiAThome replied:
Sorry for the confusion, I misremembered yung api ng better auth. For updating and deleting, it should be interfaced in using better auth api. Pero for simple read operations ok lang yung current implementation. Di lang muna implement yung deletion kasi maraming cascading tables yung mangyayari, if ever we do one, I think soft delete lang yung gagawin natin. For the updating, emails and password lang yung need mag go through betterAuth api, since through dlsu gmail lang naman pag login di na pala to needed. Name and profile pic are ok narin with the current implementation. Though wala pa tayong object storage so wala tayong malalagyan ng profile. So atleast for now name lang muna yung gawan ng update endpoint.

xandreiAThome replied:
Sorry talaga sabog lang ako kanina hahaha

xandreiAThome replied:
So just refactor it to function based implementation and remove updating of pfp lang

xandreiAThome commented on src/lib/util/auth/session.ts:
Put this under lib/util/auth to be more explicit

xandreiAThome replied:
Also create a client config for better auth so that the frontend also has these kinds of util functions. This utils is just for the server, so need rin gumawa pa ng utils for the frontend client

xandreiAThome reviewed (changes requested):
Using the auth client config create the util function na rin for signup and signout and integrate it to the login page to test for functionality. Refer [here](https://www.better-auth.com/docs/basic-usage)

rdgonzaga committed:
feat(auth): integrate BetterAuth to login page

- fix duplicate import on page.tsx
- fix deprecated generate id implementation
- rename Login.tsx to LoginModal.tsx to follow page.tsx import
- add clickable sign in button

rdgonzaga committed:
chore(auth): move session to util/auth

rdgonzaga committed:
feat(auth): add frontend client utils

rdgonzaga committed:
refactor(auth): remove AuthService methods and export session utilities

rdgonzaga committed:
refactor(auth): rename LoginModal to Login and fix imports

rdgonzaga committed:
feat(auth): add user query and update functions

rdgonzaga committed:
fix(comments): update getCommentsByArticleID to filter out replies

xandreiAThome reviewed (approved):
Looks good
